File: 07_pointers/pointers.go
package main

import "fmt"

func main() {
    var direccion *int      // declaras puntero int
    numero := 42            // int
    direccion = &numero    // direccion guarda al numero
    valor := *direccion     // dereferencial al valor

    fmt.Printf("direccion: %v\n", direccion)    // direccion
    fmt.Printf("valor: %v\n", valor)            // valor: 42
}

File: 07_pointers/pointers.c
#include <stdio.h>

int main() {
    int *direccion;
    int numero = 42;
    direccion = &numero;
    int valor = *direccion;

    printf("direccion: %p\n", direccion);
    printf("valor: %d\n", valor);

    return 0;
}
